src/tests/simulation/tests/integration_tests/backfill_tests.rs

```diff
@@ -190,6 +190,44 @@ async fn test_arrangement_backfill_replication() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_backfill_backpressure() -> Result<()> {
+    let mut cluster = Cluster::start(Configuration::default()).await?;
+    let mut session = cluster.start_session();
+
+    // Create dimension table
+    session.run("CREATE TABLE dim (v1 int);").await?;
+    // Ingest
+    // Amplification of 1000 records
+    session
+        .run("INSERT INTO dim SELECT 1 FROM generate_series(1, 200);")
+        .await?;
+    // Create fact table
+    session.run("CREATE TABLE fact (v1 int);").await?;
+    // Create sink
+    session
+        .run("CREATE SINK s1 AS SELECT fact.v1 FROM fact JOIN dim ON fact.v1 = dim.v1 with (connector='blackhole');")
+        .await?;
+    session.run("FLUSH").await?;
+
+    // Ingest
+    tokio::spawn(async move {
+        session
+            .run("INSERT INTO fact SELECT 1 FROM generate_series(1, 100000);")
+            .await
+            .unwrap();
+    })
+    .await?;
+    let mut session = cluster.start_session();
+    session.run("FLUSH").await?;
+    // Run flush to check if barrier can go through. It should be able to.
+    // There will be some latency for the initial barrier.
+    session.run("FLUSH;").await?;
+    // But after that flush should be processed timely.
+    session.run("FLUSH;").await?;
```

Comment:
I have added a timeout. Note that the join here is cpu bound, so it may take a long time in real-world, but in madsim, we don't tick the clock unless there's IO. So it may not timeout.

Regardless, if the test runtime of this test spikes, it should trigger CI timeout.

Alternatively we can run it on a non-sim environment, e.g. via `run-backfill-tests.sh` in the future.

Reply:
Indeed. I forgot it runs in madsim. Also not sure what's the clock's behavior in mad-sim.

Reply:
> but in madsim, we don't tick the clock unless there's IO. So it may not timeout.